File: flux_movie_generation/volumes/getvolumes.f90
	program getvolumes
	integer :: stat,count
	character(len=100) :: line
	character(len=100) :: orgline

	orgline="                                        unit     uses    region     mixture        total volume"
	count=0
	open(25,file="_out",status="unknown")
        open(21,file="volumes",status="unknown")
	open(22,file="ncor",status="unknown")
	do
		read(25,'(a)',iostat=stat) line
		if (  stat /= 0 ) exit
		if ( line(1:95) == orgline(1:95) ) then
			do
				read(25,'(a)',iostat=stat) line
				if (line == "" ) then
					count=count+1
					if (count > 1) exit
					cycle
				end if
				if (line(72:72) == " " ) line(72:72)="0"
				write(21,'(a)') line(80:92)
				write(22,'(a)') line(70:73)
				count=0
			end do
			if (count > 2) exit
		else
			cycle
		endif
	end do

	end program

File: flux_movie_generation/volumes/get_volumes.py
#-- Utility for gathering volumes from KENO output file
#-- AUthor: Zander Mausolff
#-- Notes: The user enters the file name they wish to get the volumes from
#--        and this program assumes the file specified is in the current directory
#-- Returns a text file called "volumes"


import re

filename= raw_input("Enter the file name containing the array: ")

input_file = open(filename, 'r+')
output_file = open('volumes','w')

i=1
for line in input_file:
	if "unit     uses    region     mixture        total region volume (cm**3)" in line:
		for line in input_file:
			if "mixture       total mixture volume (cm**3)            total mixture mass (gm)" in line:
				break
			if line !=  '\n':
				output_file.write(str(i) + line)
				i+=1
